fxa-client/src/lib.rs

```diff
@@ -498,24 +503,91 @@ impl FirefoxAccount {
         self.state.config.token_server_endpoint_url()
     }
 
-    pub fn handle_push_message(&self) {
-        panic!("Not implemented yet!")
+    /// Retrieve our own client instance metadata.
+    pub fn client_instance(&mut self) -> Result<ClientInstanceResponse> {
```

Comment:
Do we expect callers of this API to want to fetch the client-instance metadata bundle, or are they more likely going to want to access just individual fields like "get the current display name"?